src/software/simulation/ssl_simulation_protocol/ssl_simulation_protocol_manager.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+class SslSimulationProtocolManager
+{
+  public:
+    /**
+     * Sets up the initial simulation with robots, ball state, etc
+     */
+    SslSimulationProtocolManager::setupInitialSimulationState()
+
+    /**
+     * Starts the simulation. If the simulator is already running, this
+     * function does nothing.
+     */
+    void startSimulation();
+
+    /**
+     * Stops the simulation. If the simulator is already stopped, this
+     * function does nothing.
+     */
+    void stopSimulation();
+
+    /**
+     * Sets the state of the ball in the simulation. No more than 1 ball may exist
+     * in the simulation at a time. If a ball does not already exist, a ball
+     * is added with the given state. If a ball already exists, its state is set to the
+     * given state.
+     *
+     * @param ball_state The new ball state
+     */
+    void setBallState(const BallState& ball_state);
```

Comment:
would this get called by the network module?

Reply:
This would be for controlling the ball state in grsim ([proto](https://github.com/RoboCup-SSL/ssl-simulation-protocol/blob/412cdb011f59bc87bd3950d6bb72361b5c108ca5/proto/ssl_simulation_control.proto#L9)), so the implementation might be that `setBallState` uses a function in the network module to send the proto

That being said, I don't think a minimum working grsim support would need simulation control so maybe simulation control can be put as low priority